google/api/http.proto

```diff
@@ -354,6 +359,15 @@ message HttpRule {
   // message type.
   string response_body = 12;
 
+  // The name of fields whose values are mapped to HTTP headers in a response.
+  // If a field is in the response that matches a value in this map, an HTTP
```

Comment:
Same comment as above, but inverted:  what if multiple fields are mapped to the same header?  (The use case for mapping the same field value to multiple headers also seems like a stretch, but at least the behavior is better defined...)

Reply:
This issue already exists today. For example, ?a=1&b=2&a=3&b=4

The current implementation, hence the semantics, is the last one wins.

Reply:
Query params are ordered, so "last one wins" is well-defined.  But map entries are not ordered, so even if we wanted "last one wins" semantics, I'm not sure we could have them.

Inverting the maps seems to make this problem go away entirely.  Do you see any downsides to doing so?

Reply:
Can you spell out the details and let's see if it can fully address ambiguity?

Reply:
I think it's better to just state that the keys and values must all be unique here, allowing only 1-1 mappings. 

Is there really a good reason for allowing many-to-one in either direction? I can't think of one that wouldn't just lead to more esoteric behavior (e.g. if there's two mappings, which one wins?)

Reply:
Updated to require all keys/values to be unique. thoughts?

Reply:
There's protocol-level enforcement of key uniqueness, but value uniqueness needs to be implemented separately (e.g. by a linter).  Not the end of the world, of course.

> Is there really a good reason for allowing many-to-one in either direction? I can't think of one that wouldn't just lead to more esoteric behavior (e.g. if there's two mappings, which one wins?)

If you allow a header to be mapped to multiple request fields, or a response field to be mapped to multiple headers, then doesn't the "if there's two mappings, which one wins?" question go away?